net/host.go

```diff
@@ -80,17 +81,19 @@ type Config struct {
 	ProtocolID string
 	ListenIP   string
 	IsRelayer  bool
+	IsBootnode bool
 }
 
 // NewHost returns a new Host.
 // The host implemented in this package is swap-specific; ie. it supports swap-specific
 // messages (initiate and query).
 func NewHost(cfg *Config) (*Host, error) {
 	h := &Host{
-		ctx:       cfg.Ctx,
-		h:         nil, // set below
-		isRelayer: cfg.IsRelayer,
-		swaps:     make(map[types.Hash]*swap),
+		ctx:        cfg.Ctx,
+		h:          nil, // set below
+		isRelayer:  cfg.IsRelayer,
+		isBootnode: cfg.IsBootnode,
```

Comment:
This is a super short function. We could remove `IsBootnode` from this config, add a `NewBootnodeOnlyHost` method, and then `panic` if `isRelayer` is set in the new bootnode-only method.

A regular host can be a bootnode without setting `IsBootnode`, but a bootnode-only host cannot be a relayer. We want to make this as self-documenting as possible while scanning the least amount of code. Right now it is a little confusing to know how to set the values correctly and what they actually mean.

Reply:
maybe I can change the variable name to `IsBootnodeOnly` or something like that?

Reply:
Yes, that was part of the suggestion. Changing the variable name eliminates most of the confusion. The rest of the suggestion wouldn't be much extra code, especially when you consider that the existing `NewHost` and `Config` struct no longer need updates, but those changes are less important.

Reply:
I changed the variable name and added a check that `IsRelayer` and `IsBootnodeOnly` are exclusive!